src/notifications/container.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+
+import * as actions from './actions'
+import * as selectors from './selectors'
+import NotificationList from './components/NotificationList'
+import Notification from './components/Notification'
+import StatusHeading from './components/StatusHeading'
+
+class NotificationContainer extends Component {
+    static propTypes = {
+        unreadNotificationList: PropTypes.arrayOf(PropTypes.object).isRequired,
+        readNotificationList: PropTypes.arrayOf(PropTypes.object).isRequired,
+        init: PropTypes.func.isRequired,
+        showMoreIndex: PropTypes.string,
+        handleToggleShowMore: PropTypes.func.isRequired,
+        handleTick: PropTypes.func.isRequired,
+    }
+
+    componentDidMount() {
+        this.props.init()
+    }
+
+    isNotificationTruncated(message) {
+        const NotificationCharLimit = 150
+
+        if (message.length <= NotificationCharLimit) {
+            return false
+        } else {
+            return true
+        }
+    }
+
+    truncateText(message) {
+        const NotificationCharLimit = 150
+
+        const lastSpaceBeforeCutoff = message.lastIndexOf(
+            ' ',
+            NotificationCharLimit,
+        )
+        const trunctatedText = message.substr(0, lastSpaceBeforeCutoff)
+        return trunctatedText
+    }
+
+    renderNotificationItems(notifications, isUnread) {
+        const { showMoreIndex, handleToggleShowMore, handleTick } = this.props
+
+        return notifications.map((notification, i) => {
+            const isTruncated = this.isNotificationTruncated(
+                notification.message,
+            )
+
+            const trucatedMessage =
+                isTruncated && showMoreIndex !== notification.id
+                    ? this.truncateText(notification.message) + '...'
```

Comment:
Put the ellipses literal string here ('...') inside the `truncateText` method. I would also use the single ellipses unicode character rather than three periods: …